### PR TITLE
implement different chain retrieval methods

### PR DESCRIPTION
**Pull request by mikeseese**
This PR adds various methods for retrieving indexed information from the chain to support Ganache UI usage:
- `Filecoin.ChainGetTipSet`
- `Filecoin.ChainGetTipSetByHeight`
- `Filecoin.ChainGetBlock`
- `Filecoin.ChainGetBlockMessages`
- `Filecoin.ChainGetMessage`

### CODE REVIEW

**Review by tcoulter — APPROVED** [review]
All good!

File: src/chains/filecoin/filecoin/src/things/block-header.ts [hunk 1]
```diff
@@ -171,8 +171,8 @@ class BlockHeader
       timestamp: {
         deserializedName: "timestamp",
         serializedName: "Timestamp",
-        defaultValue: () => {
-          return new Date().getTime() / 1000;
+        defaultValue: literal => {
+          return literal || new Date().getTime() / 1000;
```
Comment (davidmurdoch) [inline src/chains/filecoin/filecoin/src/things/block-header.ts]: how's this work? If `literal` is a `0` somehow will this behave appropriately?

Comment (mikeseese) [reply src/chains/filecoin/filecoin/src/things/block-header.ts]: This should probably be

```suggestion
          return typeof literal !== "undefined" ? literal : (new Date().getTime() / 1000);
```

Comment (mikeseese) [reply src/chains/filecoin/filecoin/src/things/block-header.ts]: I applied the above change, but FWIW, this field is unused by Ganache

**Review by davidmurdoch — APPROVED** [review]

**Pull request merged**